lib/rbac/filterer.rb

```diff
@@ -770,9 +770,9 @@ def apply_scope(klass, scope)
     end
 
     def apply_select(klass, scope, extra_cols)
-      return scope if extra_cols.blank?
-
-      (scope.select_values.blank? ? scope.select(klass.arel_table[Arel.star]) : scope).select(extra_cols)
+      scope = scope.select(klass.arel_table[Arel.star]) if scope.select_values.blank?
```

Comment:
@kbrock So I had to rebase this PR at the request of @dmetzger57 and this spot is the only place where there is a conflict, and I know this is a method you just changed.

This is what I had previously, but what are your thoughts on this version versus the one you had?  Curious if you have an opinion here, but I am not necessarily tied to how I did things either.

Reply:
I definitely prefer the guard at the beginning of the method.
Then, this scope would not add the `Arel.star` when the extra_cols is not present.

But I'm good either way.

Reply:
@kbrock would be be fine with this:

```ruby
def apply_select(klass, scope, extra_cols)
  return scope if extra_cols.blank?

  scope = scope.select(klass.arel_table[Arel.star]) if scope.select_values.blank?
  scope.select(extra_cols)
end
```

Mostly find the current method a bit hard to parse mentally, so I am a bit partial to breaking it up a bit.  However, I do agree that the guard is a bit easier to deal with.

That said, I think they are mostly functionally equivalent, so I would be fine with not changing it at all, so maybe that route is best/safest?  I think I might refactor in a separate PR in a bit, and rebase this PR to remove any changes to this method if that sounds good to you.

Reply:
Went ahead and rebased out this change.  Will open up a separate PR, but assuming test pass, I think we are good with what was here previously.